core/domain/learner_progress_services_test.py

```diff
@@ -522,6 +597,24 @@ def test_get_all_incomplete_exp_ids(self):
             learner_progress_services.get_all_incomplete_exp_ids(
                 self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+        # Mark EXP_ID_2 as completed and create summary list to test output.
+        learner_progress_services.mark_exploration_as_completed(
```

Comment:
Can we create a separate test for testing this behavior? (Following the instruction provided in [the  wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests))

Reply:
Can you clarify what you're referring to by "this behavior"? 
Do you mean test "mark_exploration_as_completed" and "mark_collection_as_completed"?

Reply:
> # Test that completed exp summaries don't include private exploration.
The new behavior which you are testing^.

Reply:
@DubeySandeep Just to clarify, I should split up each of the 4 test functions I wrote into 2, where the FIRST one is all the code before line 611, and the SECOND one is all the code before line 611 (without the assert statements) and all the code after line 611.
Is this correct?

Reply:
Update: I understand what you were pointing out and made a new PR!